### PR TITLE
[DROOLS-6556] JSON marshalling slow due to inefficient com.fasterxml.…

### PR DESCRIPTION
**Pull request by tkobayas**
…jackson.databind.type.TypeFactory.findClass calls

- Benchmark

Retrospectively added benchmarks for https://issues.redhat.com/browse/DROOLS-6556

### CODE REVIEW

**Comment by tkobayas** [comment]
Local results:
```
Benchmark                                  (findDeserializerFirst)  Mode  Cnt    Score   Error  Units
JSONMarshallerDroolsCommandBenchmark.test                    false    ss  100  291.672 ± 2.378  ms/op
JSONMarshallerDroolsCommandBenchmark.test                     true    ss  100   12.116 ± 1.185  ms/op
```

**Comment by tkobayas** [comment]
I have added `kie-misc-benchmarks` module for benchmarks which don't fit other module.

**Review by sutaakar — APPROVED** [review]

**Review by hellowdan — COMMENTED** [review]
It looks good to me, but I'm a bit concerned about adding a module for it since it is Drools related. Have you considered adding a package inside the Drools module to create a new category? The modules looks good to me as a product separator, and Misc could be a category/package like Turtle is. What do you think?

**Comment by tkobayas** [comment]
@hellowdan  Thanks. Yeah, I thought that this is "kie-server" issue rather than drools issue. But yeah, it should fit in `drools-benchmarks`. I'll update soon.

**Comment by tkobayas** [comment]
@hellowdan  Done. It's much simpler. Thanks :)

File: drools-benchmarks/pom.xml [hunk 1]
```diff
@@ -61,6 +61,16 @@
       <groupId>org.kie</groupId>
       <artifactId>kie-pmml-dependencies</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.kie.server</groupId>
+      <artifactId>kie-server-api</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.optaplanner</groupId>
+          <artifactId>optaplanner-core</artifactId>
```
Comment (mariofusco) [inline drools-benchmarks/pom.xml:70]: I'm not understanding why do we need to add optaplanner, is it necessary?

Comment (tkobayas) [reply drools-benchmarks/pom.xml:70]: @mariofusco optaplanner is excluded here. `kie-server-api` depends on `optaplanner-core` but it causes a kie.conf problem when creating an jmh uber jar. So excluded.

Comment (tkobayas) [reply drools-benchmarks/pom.xml:70]: In other words, optaplanner7 has to apply the kie.conf namespace approach?

Comment (mariofusco) [reply drools-benchmarks/pom.xml:70]: Ah ok, sorry I misread this.

**Review by mariofusco — APPROVED** [review]

**Review by hellowdan — APPROVED** [review]

**Pull request merged**